# 01 — Análisis Exploratorio de Datos (EDA)
**CRISP-DM: Comprensión de los datos** · Etapa 1: *EDA completo (gráficos obligatorios)*

Tendencia, estacionalidad, distribución, correlaciones y outliers de los 5 productos
objetivo. Las figuras se guardan en `reports/figures/`.

In [1]:
import sys
from pathlib import Path
sys.path.append(str(Path.cwd().parent / "src"))  # importar el paquete sin instalar
import pandas as pd
from prediccion_precios import data_loading as dl, preprocessing as pp, features as ft, eda, config
pd.set_option("display.width", 160); pd.set_option("display.max_columns", 60)

In [2]:
df = dl.filtrar_productos_objetivo(dl.cargar_precios_consolidados())
dl.describir_dataset(df)

{'n_filas': 3640,
 'n_columnas': 9,
 'columnas': ['Archivo',
  'Hoja',
  'Producto',
  'Precio',
  'Fecha',
  'Dia',
  'Mes',
  'Anio',
  'Grupo'],
 'rango_fechas': (Timestamp('2021-05-07 00:00:00'),
  Timestamp('2024-06-26 00:00:00')),
 'n_fechas_unicas': 722,
 'n_productos': 5,
 'productos': ['ARROZ ORO PRIMERA CLASE IMPORTADO',
  'FRIJOL  ROJO DE SEDA NACIONAL',
  'MAÍZ BLANCO',
  'NARANJA VALENCIA MEDIANA',
  'TOMATE DE PASTA GRANDE'],
 'conteo_por_categoria': {'GranosBasicos': 2184,
  'Frutas': 728,
  'Hortalizas': 728},
 'pct_faltantes_por_columna': {'Archivo': 0.0,
  'Hoja': 0.0,
  'Producto': 0.0,
  'Precio': 0.3,
  'Fecha': 0.0,
  'Dia': 0.0,
  'Mes': 0.0,
  'Anio': 0.0,
  'Grupo': 0.0},
 'precio_describe': {'count': 3629.0,
  'mean': 36.962,
  'std': 28.095,
  'min': 7.8,
  '25%': 15.733,
  '50%': 27.704,
  '75%': 45.833,
  'max': 115.0}}

## 1. Resumen estadístico por producto

In [3]:
eda.resumen_estadistico(df)[['count','mean','std','min','max','pct_faltantes']]

,count,mean,std,min,max,pct_faltantes
Producto,,,,,,
ARROZ ORO PRIMERA CLASE IMPORTADO,726.0,43.751,3.070,38.364,46.545,0.27
FRIJOL ROJO DE SEDA NACIONAL,726.0,85.460,19.556,51.429,115.000,0.27
MAÍZ BLANCO,726.0,24.657,4.632,15.500,30.844,0.27
NARANJA VALENCIA MEDIANA,725.0,12.548,2.479,9.000,18.000,0.41
TOMATE DE PASTA GRANDE,726.0,18.358,5.893,7.800,37.727,0.27


## 2. Series temporales (tendencia)

In [4]:
for p in df['Producto'].unique():
    eda.graficar_serie_temporal(df, p)
print('Series guardadas en reports/figures/')

Series guardadas en reports/figures/


## 3. Estacionalidad (precio por mes)

In [5]:
for p in df['Producto'].unique():
    eda.graficar_estacionalidad(df, p)
print('Boxplots guardados')

Boxplots guardados


## 4. Distribución de precios

In [6]:
eda.graficar_distribucion(df)

WindowsPath('C:/Users/irmal/Downloads/prediccion_precios_agricolas_V2/prediccion/reports/figures/distribucion_precios.png')

## 5. Outliers (IQR)

In [7]:
eda.detectar_outliers(df)

,Producto,n,n_outliers,pct_outliers,limite_inf,limite_sup
0,TOMATE DE PASTA GRANDE,726,12,1.65,1.906,34.156
1,ARROZ ORO PRIMERA CLASE IMPORTADO,726,0,0.00,29.470,55.652
2,FRIJOL ROJO DE SEDA NACIONAL,726,0,0.00,21.096,148.789
3,MAÍZ BLANCO,726,0,0.00,6.781,42.109
4,NARANJA VALENCIA MEDIANA,725,0,0.00,4.000,20.000


## 6. Correlación de precios entre productos

In [8]:
eda.matriz_correlacion(df)

Producto,ARROZ ORO PRIMERA CLASE IMPORTADO,FRIJOL ROJO DE SEDA NACIONAL,MAÍZ BLANCO,NARANJA VALENCIA MEDIANA,TOMATE DE PASTA GRANDE
Producto,,,,,
ARROZ ORO PRIMERA CLASE IMPORTADO,1.000,0.910,0.809,0.282,0.317
FRIJOL ROJO DE SEDA NACIONAL,0.910,1.000,0.778,0.452,0.276
MAÍZ BLANCO,0.809,0.778,1.000,0.028,0.215
NARANJA VALENCIA MEDIANA,0.282,0.452,0.028,1.000,0.147
TOMATE DE PASTA GRANDE,0.317,0.276,0.215,0.147,1.000


### Hallazgos del EDA
- Los granos básicos se mueven muy juntos (arroz–frijol ≈ 0.91, arroz–maíz ≈ 0.81).
- Naranja y tomate son casi independientes del resto.
- Todos los productos objetivo tienen ~0.3% de faltantes tras la selección de variantes.
> Completar con observaciones de tendencia y estacionalidad por cultivo.